### PR TITLE
refactor[devtools/extension]: refactored messaging logic across different parts of the extension

### PR DESCRIPTION
**Pull request by hoxyq**
1. https://github.com/bvaughn/react/commit/9fc04eaf3fb701cdc14f57d5aed48f3126af6c94#diff-2c5e1f5e80e74154e65b2813cf1c3638f85034530e99dae24809ab4ad70d0143 introduced a vulnerability: we listen to `'fetch-file-with-cache'` event from `window` to fetch sources of the file, in which we want to parse hook names. We send this event via `window`, which means any page can also use this and manipulate the extension to perform some `fetch()` calls. With these changes, instead of transporting message via `window`, we have a distinct content script, which is responsible for fetching sources. It is notified via `chrome.runtime.sendMessage` api, so it can't be manipulated.
2. Consistent structure of messages `{source: string, payload: object}` in different parts of the extension
3. Added some wrappers around `chrome.scripting.executeScript` API in `packages/react-devtools-extensions/src/background/executeScript.js`, which support custom flow for Firefox, to simulate support of `ExecutionWorld.MAIN`.

### CODE REVIEW

File: packages/react-devtools-extensions/src/background/messageHandlers.js [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+import {EXTENSION_CONTAINED_VERSIONS} from '../utils';
+
+export function handleReactDevToolsHookMessage(message, sender) {
+  const {payload} = message;
+
+  switch (payload?.type) {
+    case 'react-renderer-attached': {
+      setExtensionIconAndPopup(payload.reactBuildType, sender.tab.id);
+    }
```
Comment (motiz88) [inline packages/react-devtools-extensions/src/background/messageHandlers.js:14]:
```suggestion

      break;
    }
```

File: packages/react-devtools-extensions/src/contentScripts/fileFetcher.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    error => reject(null),
+  );
+}
+
+chrome.runtime.onMessage.addListener(message => {
+  if (
+    message?.source === 'devtools-page' &&
+    message?.payload?.type === 'fetch-file-with-cache'
+  ) {
+    console.log('fetch start');
```
Comment (motiz88) [inline packages/react-devtools-extensions/src/contentScripts/fileFetcher.js:48]: Are these `console.log` calls intended to remain in the code?

**Review by motiz88 — APPROVED** [review]
Broadly makes sense to me - not super familiar with the code though!

File: packages/react-devtools-extensions/src/contentScripts/fileFetcher.js (hunk 2 above)
Comment (hoxyq) [reply packages/react-devtools-extensions/src/contentScripts/fileFetcher.js:48]: Oops, good catch. Was testing something and forgot to remove

**Pull request merged**